libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -234,11 +233,21 @@ AP_GPS_UBLOX::_verify_rate(uint8_t msg_class, uint8_t msg_id, uint8_t rate) {
                 _cfg_needs_save = true;
             }
             break;
+        case MSG_PVT:
+            if(rate == RATE_PVT) {
+                _unconfigured_messages &= ~CONFIG_RATE_PVT;
+            } else {
+                _configure_message_rate(msg_class, msg_id, RATE_PVT);
+                _unconfigured_messages |= CONFIG_RATE_PVT;
+                _cfg_needs_save = true;
+            }
```

Comment:
I'm inclined to agree with you, but I'd rather be conservative and wait for the unequivocal proof which is actually receiving a PVT message. It shouldn't take to long to see message show up so you will still get the proof.

The other reason I don't want to flag it as true here, is that the Tx buffer might be true and we might never get the PVT enable message out, but we would start dropping the POSLLH/VELNED messages. This would unfortunately mean we lose the driver instance before we ever reattempt to enable PVT.